autode/values.py

```diff
@@ -0,0 +1,401 @@
+        if other is None:
+            return False
+
+        if isinstance(other, Value):
+            if other.units == self.units:
+                return np.isclose(other, float(self))
+            else:
+                return np.isclose(other.to(self.units), float(self))
+
+        return np.isclose(other, float(self))
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __lt__(self, other):
+        """Less than comparison operator"""
+
+        if isinstance(other, Value):
+            return float(self) < other.to(self.units)
+
+        return float(self) < other
+
+    def __gt__(self, other):
+        """Greater than comparison operator"""
+        return not self.__lt__(other)
+
+    def __le__(self, other):
+        """Greater than or equal to comparison operator"""
+        return self.__lt__(other) or self.__eq__(other)
+
+    def __ge__(self, other):
+        """Less than or equal to comparison operator"""
+        return self.__gt__(other) or self.__eq__(other)
+
+    def __add__(self, other):
+        """Add another value onto this one"""
+        if isinstance(other, np.ndarray):
+            return other + float(self)
+
+        return self.__class__(float(self) + self._other_same_units(other),
+                              units=self.units)
+
+    def __mul__(self, other):
+        """Multiply this value with another"""
+        if isinstance(other, np.ndarray):
+            return other * float(self)
+
+        return self.__class__(float(self) * self._other_same_units(other),
+                              units=self.units)
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __radd__(self, other):
+        return self.__add__(other)
+
+    def __sub__(self, other):
+        return self.__add__(-other)
+
+    def to(self, units):
+        """Convert this value to a new unit, returning a copy
+
+        Arguments:
+            units (autode.units.Unit | str):
+
+        Returns:
+            (autode.values.Value):
+
+        Raises:
+            (TypeError):
+        """
+        return _to(self, units)
+
+    def __init__(self, x,
+                 units: Union[Unit, None] = None):
+        """
+        Value constructor
+
+        Arguments:
+            x (float | int):
+
+        Keyword Arguments:
+            units (autode.units.Unit | None):
+        """
+        float.__init__(float(x))
+        self.units = units
+
+
+class Energy(Value):
+    """Type of energy in some units e.g. Potential, Free etc.
+    defaults to Hartrees"""
+
+    implemented_units = [ha, kcalmol, kjmol, ev]
+
+    def __str__(self):
+        return f'Energy({round(self, 5)} {self.units.name})'
+
+    def __eq__(self, other):
+        """Is an energy equal to another? Compares only the value, with
+        implicit unit conversion"""
+
+        # A PotentialEnergy is not equal to a FreeEnergy, for example
+        if isinstance(other, Energy) and not isinstance(other, self.__class__):
```

Comment:
is the isinstance(other, Energy) not covered by the second part?

Reply:
Not correctly I don't think, because this needs both

```
Energy(1.0) == 1
PotentialEnergy(1.0) != FreeEnergy(1.0)
```